### ISSUE TITLE
No entries made in db.topology for server address: '1666', dest address: '' and serverID: ''.

### ISSUE THREAD

**Issue by JasonMa0012** [FrozenStormInteractive/Perforce-Synology#2]
System: DSM 7.2.1-69057 Update 5
Release: HelixCoreServer-24.1-0041-x86_64-7.2.spk

I installed the perforce server on /volume3 and P4ROOT on /volume1. I had some problems with the first installation, but I eventually solved them, logged in successfully in p4v on Windows, and uploaded some files.

Then I realised that using the p4d command was always accompanied by the error: `Perforce password (P4PASSWD) invalid or unset.`, and the error persisted even after logging in.

After analysing the problem I found out that it was caused by not having write access to P4TICKETS, but I didn't know how to fix it, so I deleted the perforce server as you mentioned in #1 and reinstalled it: 

```

XXX@XXX:~$ export P4PASSWD=XXX
XXX@XXX:~$ cd /volume1/p4
XXX@XXX:/volume1/p4$ sudo -u Perforce p4dctl-ng new
Password: XXX
✔ Perforce Service name: jasonmap4
✔ Perforce Server root (P4ROOT): /volume1/p4/jasonmap4
✔ Perforce Server unicode-mode: Yes
✔ Perforce Server case-sensitive (Not recommended for compatibility with Windows clients): No
✔ Perforce Server address (P4PORT): 1234
✔ Perforce super-user login: jasonma
✔ Perforce super-user password: XXX
✔ Initialize the server with the recommended settings: Yes
✔ Start the server after the configuration: Yes
✔ Service config path (where to save the .conf file): /var/packages/HelixCoreServer/etc/p4dctl-ng.conf.d/jasonmap4.conf
Configuring p4d service 'jasonmap4' with the information you specified...
Service conf file saved as /var/packages/HelixCoreServer/etc/p4dctl-ng.conf.d/jasonmap4.conf
Enabling unload and spec depots
Populating the typemap
Initializing protections table...
Error when initializing the server: Perforce password (P4PASSWD) invalid or unset.

```

You can see that even though I set the password in advance using `export P4PASSWD=XXX`, I still get `Perforce password (P4PASSWD) invalid or unset`.

At this point, even though the database has been successfully created at ` /volume1/p4/jasonmap4`, attempting to start the service will result in the error:

```

XXX@XXX:/volume1/p4$ p4d
Perforce db files in '.' will be created if missing...
Perforce Server starting...
Perforce server warning:
        Pid 26185
        Operation: topologyRegistration
        No entries made in db.topology for server address: '1666', dest address: '' and serverID: ''.
        ServerID for the server should be set and a server restart is required.

```

Additional information is as follows:

```

XXX@XXX:/volume1/p4$ p4 info
User name: XXX
Client name: jasonmap4
Client host: jasonmap4
Client unknown.
Current directory: /volume1/p4
Peer address: 127.0.0.1:60796
Client address: 127.0.0.1
Server address: localhost:2333
Server root: /volume1/p4/jasonmap4/root
Server date: 2024/08/29 22:17:14 +0800 CST
Server uptime: 00:00:55
Server version: P4D/LINUX26X86_64/2024.1/2625008 (2024/07/15)
ServerID: jasonmap4
Server services: standard
Server license: none
Case Handling: insensitive


XXX@XXX:/volume1/p4$ sudo cat /var/packages/HelixCoreServer/etc/p4dctl-ng.conf.d/jasonmap4.conf
Password: XXX
{
  "Name": "jasonmap4",
  "ServerType": "p4d",
  "Owner": null,
  "Execute": "/volume3/@appstore/HelixCoreServer/usr/local/bin/p4d",
  "Enabled": true,
  "Environment": {
    "P4ROOT": "/volume1/p4/jasonmap4/root",
    "P4PORT": "1234",
    "P4SSLDIR": "/volume1/p4/jasonmap4/root/ssl"
  }

```

The problem is still reproducible after deleting and installing again, am I doing something wrong?

**Comment by JasonMa0012** [comment]
I also noticed something strange, some of the files in `/volume1/p4/jasonmap4/root` appear in `/volume1/p4`, which doesn't seem right, since I didn't have this problem when I first installed to `/volume1/Perforce/jasonmap4`

![image](https://github.com/user-attachments/assets/bea3baef-d6ae-4062-be39-23f5488c12c8)
![image](https://github.com/user-attachments/assets/8363e6ba-019d-426a-8f4a-410fc253231d)

**Comment by JasonMa0012** [comment]
Hi! I found that the service is actually running successfully.

- `No entries made in db.topology for server address: '1666', dest address: '' and serverID: ''. `It's just because I'm calling p4d incorrectly.
- `Error when initializing the server: Perforce password (P4PASSWD) invalid or unset.` This is probably due to special characters in the password, and will cause the password to be changed when logging in from p4v, and then get stuck.

After I installed P4ROOT to `/volume1/Perforce/XXXX` with a password that does not contain special characters, everything works fine!
